build.rs

```diff
@@ -273,23 +270,23 @@ fn convert_elf_to_bin(elf_path: PathBuf) -> PathBuf {
 
 // dummy implementations because docsrs builds have no network access
 
-#[cfg(docsrs_dummy_build)]
+#[cfg(any(docsrs_dummy_build, not(feature = "uefi")))]
 fn build_uefi_bootloader(_out_dir: &Path) -> PathBuf {
     PathBuf::new()
 }
-#[cfg(docsrs_dummy_build)]
+#[cfg(any(docsrs_dummy_build, not(feature = "bios")))]
 fn build_bios_boot_sector(_out_dir: &Path) -> PathBuf {
     PathBuf::new()
 }
-#[cfg(docsrs_dummy_build)]
+#[cfg(any(docsrs_dummy_build, not(feature = "bios")))]
 fn build_bios_stage_2(_out_dir: &Path) -> PathBuf {
     PathBuf::new()
 }
-#[cfg(docsrs_dummy_build)]
+#[cfg(any(docsrs_dummy_build, not(feature = "bios")))]
 fn build_bios_stage_3(_out_dir: &Path) -> PathBuf {
     PathBuf::new()
 }
-#[cfg(docsrs_dummy_build)]
+#[cfg(any(docsrs_dummy_build, not(feature = "bios")))]
 fn build_bios_stage_4(_out_dir: &Path) -> PathBuf {
     PathBuf::new()
 }
```

Comment:
Why do we need to modify the feature gates here?

Reply:
If one of the features is disabled, for example `uefi`, then `build_uefi_bootloader` is undefined. It is used in the `main` function in build.rs.

A different solution would be to do this, so the `build_*_bootloader` functions can be left undefined:
```rs
    #[cfg(feature = "uefi")]
    {
        let uefi_path = build_uefi_bootloader(&out_dir);
        println!(
            "cargo:rustc-env=UEFI_BOOTLOADER_PATH={}",
            uefi_path.display()
        );
    }

    #[cfg(feature = "bios")]
    {
        let bios_boot_sector_path = build_bios_boot_sector(&out_dir);
        println!(
            "cargo:rustc-env=BIOS_BOOT_SECTOR_PATH={}",
            bios_boot_sector_path.display()
        );
        let bios_stage_2_path = build_bios_stage_2(&out_dir);
        println!(
            "cargo:rustc-env=BIOS_STAGE_2_PATH={}",
            bios_stage_2_path.display()
        );

        let bios_stage_3_path = build_bios_stage_3(&out_dir);
        println!(
            "cargo:rustc-env=BIOS_STAGE_3_PATH={}",
            bios_stage_3_path.display()
        );

        let bios_stage_4_path = build_bios_stage_4(&out_dir);
        println!(
            "cargo:rustc-env=BIOS_STAGE_4_PATH={}",
            bios_stage_4_path.display()
        );
    }
```

Which do you prefer?

Reply:
I think I prefer the second variant, i.e. the additional feature gates in `main`. It's easier to understand what's going on this way in my opinion.